src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -592,6 +593,15 @@ public String getDatasetApprovedUsersContent(AuthUser authUser, Integer datasetI
         return builder.toString();
     }
 
+    public Collection<User> getUsersApprovedForDataset(Dataset dataset) {
+        List<DataAccessRequest> dars = this.dataAccessRequestDAO.findAllApprovedDataAccessRequestsByDatasetId(dataset.getDataSetId().toString());
+        Set<Integer> userIds = dars.stream().map(DataAccessRequest::getUserId).collect(Collectors.toSet());
+        if (userIds.isEmpty()) {
+            return List.of();
+        }
```

Comment:
If instead you just queried for the `dar.user_id` instead of all of the related DAR properties, you could simplify this quite a bit.